### PR TITLE
BUGFIX: Handle response errors in api requests

### PR DESCRIPTION
**Pull request by Sebobo**
We sometimes have the case that the prunner doesn't send proper responses and the `json_decode` method returns null instead of an array which leads to errors in the `PipelinesAndJobsResponse::fromArray`. When the response is first checked and a runtime exception is thrown this can be handled for the user.

### CODE REVIEW

**Comment by Sebobo** [comment]
This is one option. I'm not quite sure what the status code in our case was as I couldn't debug on the problematic system.

File: Classes/PrunnerApiService.php [hunk 1]
```diff
@@ -54,13 +54,19 @@ public function loadPipelinesAndJobs(): PipelinesAndJobsResponse
...
     public function loadJobDetail(JobId $jobId): ?Job
     {
         $resultString = $this->apiCall('GET', 'job/detail?' . http_build_query(['id' => $jobId->getId()]), null)->getBody()->getContents();
         $result = json_decode($resultString, true);
+        if ($result === null) {
+            throw new \RuntimeException('Could not decode JSON response from prunner API');
+        }
         if (isset($result['error'])) {
             return null;
         }
```
Comment (mhsdesign) [inline Classes/PrunnerApiService.php]: shouldnt we prefer `JSON_THROW_ON_ERROR` for a more meaning full exception?

Comment (Sebobo) [reply Classes/PrunnerApiService.php]: Will test if it results in the same

Comment (Sebobo) [reply Classes/PrunnerApiService.php]: Adjusted it but wrapped the exception into another one to make it better understandable.

**Comment by Sebobo** [comment]
Example:

Bildschirmfoto 2024-02-09 um 14 42 45

**Review by mhsdesign — APPROVED** [review]
gratias

**Pull request merged**